src/mlpack/core/tree/rectangle_tree/rectangle_tree_impl.hpp

```diff
@@ -419,6 +441,58 @@ RectangleTree<DistanceType, StatisticType, MatType, SplitType, DescentType,
     delete dataset;
 }
 
+// Insert a point into the dataset.
+template<typename DistanceType,
+         typename StatisticType,
+         typename MatType,
+         typename SplitType,
+         typename DescentType,
+         template<typename> class AuxiliaryInformationType>
+template<typename InMatType>
+void RectangleTree<DistanceType, StatisticType, MatType, SplitType, DescentType,
+                   AuxiliaryInformationType>::Insert(const InMatType& points)
+{
+  if (parent != NULL)
+    throw std::runtime_error("RectangleTree::Insert(): cannot insert points "
+        "into a node that is not the root of the tree!");
+
+  const size_t oldCols = dataset->n_cols;
+  dataset->insert_cols(dataset->n_cols, points);
+  for (size_t i = 0; i < points.n_cols; ++i)
+    InsertPoint(oldCols + i);
```

Comment:
So this function is only a proper interface to the old `InsertPoint` ?

Reply:
Exactly, it is a far more convenient way than manually adding the point and then calling `InsertPoint()`.